### PR TITLE
binary-0.8.0.0 will provide Binary Version

### PR DESCRIPTION
**Pull request by phadej**
Related to https://github.com/haskell/cabal/issues/2919

Maybe we should merge this on top of whatever is the last release?

See https://github.com/kolmodin/binary/pull/87

### CODE REVIEW

**Comment by 23Skidoo** [comment]
This seems to break bootstrapping.

**Comment by phadej** [comment]
ping @23Skidoo , @dcoutts 

I have no better idea how to make bootstrapping fail-proof-er; except for going bananas with template haskell.

File: Cabal/Distribution/Version.hs [hunk 1]
```diff
@@ -3,7 +3,25 @@
 #if __GLASGOW_HASKELL__ < 707
 {-# LANGUAGE StandaloneDeriving #-}
 #endif
+
+-- Hack approach to support bootstrapping
+-- Assume binary <0.8 when MIN_VERSION_binary marco is not available.
+-- Starting with GHC>=8.0, compiler will hopfully provide this macros too.
```
Comment (23Skidoo) [inline Cabal/Distribution/Version.hs]: s/hopfully/hopefully/

**Comment by 23Skidoo** [comment]
LGTM, merging. Thanks!

**Pull request merged**